src/pixel/pixel_component.rs

```diff
@@ -0,0 +1,40 @@
+use num_traits::{Num, NumAssign, NumCast, NumOps};
```

Comment:
is it possible to make num_traits optional? "Zero dependencies" is an important badge of honour for some users.

Reply:
Possibly, but at the cost of general usability in some cases when attempting math with pixel components. I'm not too worried about it personally since `num_traits` is a well-regarded crates which itself has zero-dependencies, but I don't feel too strongly about this either way.